src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7466,8 +7448,15 @@ private bool IsUsingAliasInScope(string name)
                         return BindMemberOfType(node, right, rightName, rightArity, indexed, boundLeft, typeArgumentsSyntax, typeArguments, lookupResult, BoundMethodGroupFlags.SearchExtensionMethods, diagnostics: diagnostics);
                     }
 
-                    return MakeBoundMethodGroupAndCheckOmittedTypeArguments(boundLeft, rightName, typeArguments, lookupResult,
+                    BoundExpression result = MakeBoundMethodGroupAndCheckOmittedTypeArguments(boundLeft, rightName, typeArguments, lookupResult,
                         flags: BoundMethodGroupFlags.SearchExtensionMethods, node, typeArgumentsSyntax, diagnostics);
+
+                    if (!invoked)
```

Comment:
>if (!invoked) [](http://example.com/codeflow?start=20&length=13)

I am not sure why do we do this conditionally. Is there a specific reason to keep non-methods unresolved for invocation scenarios?

Reply:
We need arguments to resolve invocation scenarios, even if we're only interested in invocable non-method members.

Consider a classic extension method in an inner scope and an extension type with an invocable non-method member in an outer scope.
Depending on whether the extension method is applicable or not, the invocable non-method member may be found or not. 
But to determine whether it is applicable, we need arguments.

Reply:
From offline discussion, it would be possible to do this unconditionally, but it would not always perform the resolution to a non-extension member (sometimes that will be possible to do only once we have arguments to discard some extension methods that are in closer scopes) and it would probably not be more readable/understandable.